cpp/references/data_structures/data_exporter.cpp

```diff
@@ -0,0 +1,57 @@
+#include "data_exporter.hpp"
+
+using namespace godot;
+
+Variant IDataExporter::_get_data() const { return Variant(); }
+
+void IDataExporter::_set_data(Variant) {}
+
+Variant IDataExporter::_get(String p_property) {
+	Variant vdata = _get_data();
+	ERR_FAIL_COND_V(vdata.get_type() != Variant::OBJECT, false);
+
+	Ref<Reference> instance = vdata;
+	ERR_FAIL_COND_V(instance.is_valid(), false);
+
+	return instance->get(p_property);
+}
+
+bool IDataExporter::_set(String p_property, Variant p_value) {
+	Variant vdata = _get_data();
+	ERR_FAIL_COND_V(vdata.get_type() != Variant::OBJECT, false);
+
+	Ref<Reference> instance = vdata;
+	ERR_FAIL_COND_V(instance.is_valid(), false);
+
+	instance->set(p_property, p_value);
+	return true;
+}
+
+Array IDataExporter::_get_property_list() {
+	Array ret;
+	Variant vdata = _get_data();
+	ERR_FAIL_COND_V(vdata.get_type() != Variant::OBJECT, ret);
```

Comment:
Not sure if I'm the one who told you this, but in this case, you wouldn't want to exit with an error message just because "data" isn't an object. If the user decided to store integers in their vertices/arcs, then we'd end up getting error messages when trying to view the object because it's not an object. We should just only append the object's property list if it is a valid reference to an object. Otherwise, don't append anything. Same goes for the `_set` and `_get` methods.